internal/ethapi/api.go

```diff
@@ -615,9 +608,9 @@ func (s *PublicBlockChainAPI) doCall(ctx context.Context, args CallArgs, blockNr
 			return "0x", gas, err
 		}
 		return common.ToHex(res), gas, err
+	} else {
+		return "0x", common.Big0, err
 	}
```

Comment:
Ah I see, you need it for the error. Then Please do the error handling first and then the long code (outside of the if still!)